pkg/common/configs/config_test.go

```diff
@@ -907,15 +907,15 @@ partitions:
 	conf, err := CreateConfig(data)
 	assert.NilError(t, err, "partition user parsing should not have failed")
 	// gone through validation: 1 top level queues
-	if len(conf.Partitions[0].Queues) != 1 {
-		t.Errorf("failed to load queue %v", conf)
+	if len(conf.Partitions[0].Queues) != 1 || len(conf.Partitions[0].Queues[0].Limits) != 3 {
+		t.Errorf("partition limits not correctly applied to root queue %v", conf)
 	}
-	if len(conf.Partitions[0].Limits) != 3 {
-		t.Errorf("partition users linked not correctly loaded  %v", conf)
+	if len(conf.Partitions[0].Limits) != 0 {
+		t.Errorf("partition limits not wiped out %v", conf)
 	}
 
 	// limit 1 check
-	limit := conf.Partitions[0].Limits[0]
+	limit := conf.Partitions[0].Queues[0].Limits[0]
```

Comment:
ok, that changes are the side effect.

or could we accept the root-level limits is equal with partition-level limits? that may be a good way, as the caller can call either root-level limits or partition-level limits.

Reply:
> or could we accept the root-level limits is equal with partition-level limits? that may be a good way, as the caller can call either root-level limits or partition-level limits.

I've considered this before, and at that time, I thought making configuration changes was a relatively simple approach without the need to modify manager.go. Anyway, this is a completely different approach. I will submit another PR and let the community decide.

Reply:
ummm, you are right. please ignore my previous comments :)